azurerm/internal/services/apimanagement/api_management_custom_domain_resource.go

```diff
@@ -0,0 +1,325 @@
+package apimanagement
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/services/apimanagement/mgmt/2019-12-01/apimanagement"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/tf"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+var apiManagementCustomDomainResourceName = "azurerm_api_management_custom_domain"
+
+func resourceArmApiManagementCustomDomain() *schema.Resource {
+	return &schema.Resource{
+		Create: apiManagementCustomDomainCreateUpdate,
+		Read:   apiManagementCustomDomainRead,
+		Update: apiManagementCustomDomainCreateUpdate,
+		Delete: apiManagementCustomDomainDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(30 * time.Minute),
+			Read:   schema.DefaultTimeout(5 * time.Minute),
+			Update: schema.DefaultTimeout(30 * time.Minute),
+			Delete: schema.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"resource_group_name": azure.SchemaResourceGroupName(),
+
+			"api_management_name": azure.SchemaApiManagementName(),
```

Comment:
I think we should use the ID of the api_management resource here, rather than the full schema.  The custom_domain resource is a meta resource that modifies an existing api_management resource, so should be intrinsically linked to it. (we can then get the name and resource group from the ID)

Reply:
All the other subresources refer to the APIM by resource_group_name and api_management_name, but if you think direct ID is better (i.e. `api_management_id`) I'll change it to that.

Reply:
By looking closer, the SDK client also expects resource group name and name parameters, meaning it would bring additional parsing with it if we switch to `api_management_id`.

Reply:
@sirlatrom - i agree with @jackofallops, we can use the id and parse out the resource group and name with a custom parse function (which this resource should be switched over to anyways) for example see pardse.CdnEndpointID

Reply:
I've done it inside the Read function for now. If you think it warrants its own parse function I can make that during the week.

Reply:
@katbyte I've made the parse function in e61e67d.